pkg/api/dashboard: Remove welcome panel on provisioned home dashboard

amnonh opened the pull request:
It is possible to set the default home dashboard from the configuration.
    
If someone sets the home dashboard, there is no need to add the getting started panel.

After this patch, the blue getting started panel is no longer part of the home dashboard.

Fixes #26447

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=28710) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=28710) before we can accept your contribution.<br/>**19** out of **20** committers have signed the CLA.<br/><br/>:white_check_mark: grafanabot<br/>:white_check_mark: bergquist<br/>:white_check_mark: zoltanbedi<br/>:white_check_mark: torkelo<br/>:white_check_mark: aknuds1<br/>:white_check_mark: aocenas<br/>:white_check_mark: elliotpryde<br/>:white_check_mark: jackw<br/>:white_check_mark: KonanMentor<br/>:white_check_mark: nonamef<br/>:white_check_mark: papagian<br/>:white_check_mark: hugohaggmark<br/>:white_check_mark: dprokop<br/>:white_check_mark: marefr<br/>:white_check_mark: Clarity-89<br/>:white_check_mark: maknik<br/>:white_check_mark: kylebrandt<br/>:white_check_mark: kwarunek<br/>:white_check_mark: wbrowne<br/>:x: Amnon Heiman<br/><hr/>**Amnon Heiman** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=28710) it.</sub>

amnonh committed:
pkg/api/dashboard: Add the welcome panels only for default home

A user can set the home dashboard, in this case there is no need
to add the welcome panels to it.

This patch check if the home dashboard was configured by the user if it
was it will not add the welcome panels.

Fixes #26447

Signed-off-by: Amnon Heiman <amnon@scylladb.com>

aknuds1 commented on pkg/api/dashboard.go:
```diff
@@ -336,8 +336,10 @@ func (hs *HTTPServer) GetHomeDashboard(c *models.ReqContext) Response {
 	}
 
 	filePath := hs.Cfg.DefaultHomeDashboardPath
+	var use_default_path bool = false
```
```suggestion
	useDefaultPath := false
```

aknuds1 commented on pkg/api/dashboard.go:
```diff
@@ -336,8 +336,10 @@ func (hs *HTTPServer) GetHomeDashboard(c *models.ReqContext) Response {
+	var use_default_path bool = false
 	if filePath == "" {
 		filePath = filepath.Join(hs.Cfg.StaticRootPath, "dashboards/home.json")
+		use_default_path = true
```
```suggestion
		useDefaultPath = true
```

aknuds1 commented on pkg/api/dashboard.go:
```diff
@@ -356,7 +358,7 @@ func (hs *HTTPServer) GetHomeDashboard(c *models.ReqContext) Response {
 	}
 
-	if c.HasUserRole(models.ROLE_ADMIN) && !c.HasHelpFlag(models.HelpFlagGettingStartedPanelDismissed) {
+	if use_default_path && c.HasUserRole(models.ROLE_ADMIN) && !c.HasHelpFlag(models.HelpFlagGettingStartedPanelDismissed) {
```
```suggestion
	if useDefaultPath && c.HasUserRole(models.ROLE_ADMIN) && !c.HasHelpFlag(models.HelpFlagGettingStartedPanelDismissed) {
```

aknuds1 committed:
Apply suggestions from code review

aknuds1 commented:
I fixed variable naming and committed it to see if it would trigger CI with the correct base branch, and it did help.

amnonh commented:
Thanks @aknuds1

torkelo commented:
Sorry missed this PR, fixed by https://github.com/grafana/grafana/pull/28750
